pkg/agent/cniserver/sriov.go

```diff
@@ -0,0 +1,130 @@
+package cniserver
+
+import (
+	"fmt"
+	"context"
+	"net"
+	"os"
+	"path"
+	"time"
+	"google.golang.org/grpc"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"antrea.io/antrea/pkg/agent/util"
+	podresourcesv1alpha1 "k8s.io/kubelet/pkg/apis/podresources/v1alpha1"
+	sriovcniutils "github.com/k8snetworkplumbingwg/sriov-cni/pkg/utils"
+	"k8s.io/klog/v2"
+)
+
+const (
+	kubeletPodResourcesPath = "/var/lib/kubelet/pod-resources"
+	kubeletSocket           = "kubelet.sock"
+	connectionTimeout       = 10 * time.Second
+	maxMsgSize              = 1024 * 1024 * 16 // 16 Mb
+)
+
+type KubeletPodResources struct {
+	resources []*podresourcesv1alpha1.PodResources
+}
+
+func GetPodResourceRequests(podName string, podNamespace string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), connectionTimeout)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, path.Join(kubeletPodResourcesPath, kubeletSocket),
+		grpc.WithInsecure(), grpc.WithContextDialer(func(ctx context.Context, addr string) (conn net.Conn, e error) {
+			return util.DialLocalSocket(addr)
+		}))
+	if err != nil {
+		return "", fmt.Errorf("error getting the gRPC client for Pod resources: %v", err)
+	}
+
+	defer conn.Close()
+
+	client := podresourcesv1alpha1.NewPodResourcesListerClient(conn)
+	if client == nil {
+		return "", fmt.Errorf("Error getting the lister client for Pod resources.")
+	}
+
+	podResources, err := client.List(ctx, &podresourcesv1alpha1.ListPodResourcesRequest{})
+	if err != nil {
+		return "", fmt.Errorf("Error getting the Pod resources: %v %v", podResources, err)
+	}
+
+	var kpr KubeletPodResources
+	kpr.resources = podResources.GetPodResources()
+	podDeviceIDs := ""
+	for _, pr := range kpr.resources {
+		if pr.Name == podName && pr.Namespace == podNamespace {
+			for _, ctr := range pr.Containers {
+				for _, dev := range ctr.Devices {
+					podDeviceIDs += fmt.Sprint(dev.DeviceIds)
+				}
+				if len(podDeviceIDs) > 2 {
+					// example: [0000:81:02.0 0000:81:0a.3]
+					// remove "[]" and return DeviceIds
+					length := len(podDeviceIDs) - 1
+					podDeviceIDs = podDeviceIDs[1:length]
+					klog.V(2).Info("Pod DeviceIds of ", podName, podNamespace, "are: ", podDeviceIDs)
+					return podDeviceIDs, nil
+				}
+			}
+		}
+	}
+
+	return "", nil
+}
+
+// getVFInfo takes in a VF's PCI device ID and returns its PF and VF ID.
+func getVFInfo(vfPCI string) (string, int, error) {
+	var vfID int
+
+	pf, err := sriovcniutils.GetPfName(vfPCI)
+	if err != nil {
+		return "", vfID, err
+	}
+
+	vfID, err = sriovcniutils.GetVfid(vfPCI, pf)
+	if err != nil {
+		return "", vfID, err
+	}
+
+	return pf, vfID, nil
+}
+
+// getVFLinkName returns VF's network interface name given it's PCI address.
```

Comment:
```suggestion
// getVFLinkName returns a VF's network interface name given its PCI address.
```

Reply:
Missed "its"; will fix it along with any other suggestions. Thanks